test-case/check-userspace-parecord.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+##
+## Case Name: check-userspace-parecord
+## Preconditions:
+##    N/A
+## Description:
+##    Go through all the sources and parecord on the sources whose
+##    active port is available (jack connected) or unknown (DMIC)
+## Case step:
+##    1. go through all the sources
+##    2. check the source's active port is available or not
+##    3. parecord on the active port which is available or unknown
+## Expect result:
+##    parecord on the source successfully
+##
+
+# shellcheck source=case-lib/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/lib.sh
+
+OPT_OPT_lst['r']='round'     OPT_DESC_lst['r']='round count'
+OPT_PARM_lst['r']=1         OPT_VALUE_lst['r']=3
+
+OPT_OPT_lst['d']='duration' OPT_DESC_lst['d']='parecord duration in second'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=8
+
+OPT_OPT_lst['f']='file'   OPT_DESC_lst['f']='source file path'
+OPT_PARM_lst['f']=1         OPT_VALUE_lst['f']='/dev/zero'
+
+OPT_OPT_lst['F']='format'   OPT_DESC_lst['F']='sample format'
+OPT_PARM_lst['F']=1         OPT_VALUE_lst['F']=s16le
+
+OPT_OPT_lst['R']='rate'   OPT_DESC_lst['R']='sample rate'
+OPT_PARM_lst['R']=1         OPT_VALUE_lst['R']=44100
+
+OPT_OPT_lst['C']='channels'   OPT_DESC_lst['C']='channels'
+OPT_PARM_lst['C']=1         OPT_VALUE_lst['C']=2
+
+func_opt_parse_option "$@"
+
+round_cnt=${OPT_VALUE_lst['r']}
+duration=${OPT_VALUE_lst['d']}
+file=${OPT_VALUE_lst['f']}
+format=${OPT_VALUE_lst['F']}
+rate=${OPT_VALUE_lst['R']}
+channel=${OPT_VALUE_lst['C']}
+
+if [[ ! -e $file ]]; then
+    dlogw "$file does not exist, use /dev/null as dummy capture output"
+    file=/dev/null
+fi
+
+# TODO: check the parameter is valid or not
+
+# go through all the sources
+# get all the sources name
+sourcekeys=$(pactlinfo.py | jq '.Source'|jq 'keys[]')
```

Comment:
do we install jq on all DUTs?

Reply:
No, we don't need install jq on all DUTs. It is only needed on user space DUTs. But I will write a patch to install jq on all DUTs.